### PR TITLE
core: fix bug where deadlock detection was always on for expiration and quotas

### PR DESCRIPTION
**Pull request by jasonodonnell**
PR #18604 introduced a bug where deadlock detection was always on for both expiration and quota subsystems. This can result in negative server performance as deadlock detection adds overhead to locks.

Deadlock detection can now be toggled on expiration and quotas via `detect_deadlocks="statelock,expiration,quotas"`.

### CODE REVIEW

**Review by ncabatoff — APPROVED** [review]
Missing changelog but otherwise seems good to go.

File: vault/core.go [hunk 1]
```diff
@@ -947,19 +951,26 @@ func CreateCore(conf *CoreConfig) (*Core, error) {
...
-	}
 
 	effectiveSDKVersion := conf.EffectiveSDKVersion
 	if effectiveSDKVersion == "" {
 		effectiveSDKVersion = version.GetVersion().Version
 	}
 
+	var detectDeadlocks []string
+	if conf.DetectDeadlocks != "" {
+		conf.DetectDeadlocks = strings.Trim(conf.DetectDeadlocks, " ")
```
Comment (hghaf099) [inline vault/core.go:962]: How about using `strings.TrimSpace` instead? Also, should we convert the given config to lower letters? 
`strings.TrimSpace(strings.ToLower(conf.DetectDeadlocks))`

Comment (raskchanky) [inline vault/core.go:962]: Over-the-top-nitpick-that-you-shouldn't-at-all-feel-pressured-to-include-but-my-over-the-top-pedantry-compelled-me-to-include: `strings.TrimSpace()` might be a bit more thorough in this case.

**Review by raskchanky — APPROVED** [review]

File: vault/core.go (hunk 1 above)
Comment (jasonodonnell) [reply vault/core.go]: Done, thanks!

Comment (jasonodonnell) [reply vault/core.go]: Done, thanks!

**Review by hghaf099 — APPROVED** [review]

**Pull request merged**